test/images/sample-device-plugin/sampledeviceplugin.go

```diff
@@ -66,6 +66,12 @@ func stubAllocFunc(r *pluginapi.AllocateRequest, devs map[string]pluginapi.Devic
 				ContainerPath: fpath,
 				HostPath:      fpath,
 			})
+
+			// add the CDI device ID to the response.
+			cdiDevice := &pluginapi.CDIDevice{
+				Name: fmt.Sprintf("cdi.%s=%s", resourceName, dev.ID),
+			}
+			response.CdiDevices = append(response.CdiDevices, cdiDevice)
```

Comment:
yes, this is a good start in the right direction. I think we should also:
1. make it possible for the plugin to optionally return empty data (depending on e2e test scenario)
2. keep and use a old version of the plugin without the field at all - this can be done just keeping the old `sample-device-plugin` image in the k8s registry, which should be the case by default

With these two changes in places we should be enabled to test all the major scenarios e2e.

Reply:
Yes, I started looking at the e2e testing and noted that quite a bit of work is required to cover the expected use cases for proper coverage. For now, I thought I would add this field to simulate a device plugin that supports this field communicating with a kubelet that doesn't yet support it -- or has the feature turned off.

On the other hand, it may make more sense to do this in a follow-up that specifically targets e2e testing instead of including this here. Do you have thoughts there?

Reply:
I think is better to do everything in a separate (set of) PR(s).
In general I think e2e tests are _desirable_, but not required at this stage. I think the best way forward is focus on good unit test coverage, file an issue or any form of tracking about adding e2e tests for this feature and point this out in the PR. Adding e2e test should be a Beta blocker, and we should keep working on this in this cycle or in the next one, alongside the other activities. 
Let's see what other reviewers think and if this path forward is reasonnable.

Reply:
OK. Let me roll-back the e2e testing changes then as part of this PR and then start looking at follow ups. I have added some better unit test coverage with the last iteration.